tls/extensions/s2n_server_server_name.c

```diff
@@ -18,11 +18,42 @@
 #include "tls/s2n_connection.h"
 #include "tls/extensions/s2n_server_server_name.h"
 
-#define s2n_server_can_send_server_name(conn) ((conn)->server_name_used && \
-        !s2n_connection_is_session_resumed((conn)))
+static bool s2n_server_name_should_send(struct s2n_connection *conn);
+static int s2n_server_name_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_server_name_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
+
+const s2n_extension_type s2n_server_server_name_extension = {
+    .iana_value = TLS_EXTENSION_SERVER_NAME,
+    .is_response = true,
+    .send = s2n_server_name_send,
+    .recv = s2n_server_name_recv,
+    .should_send = s2n_server_name_should_send,
+    .if_missing = s2n_extension_noop_if_missing,
+};
+
+static bool s2n_server_name_should_send(struct s2n_connection *conn)
+{
+    return conn && conn->server_name_used && !s2n_connection_is_session_resumed(conn);
+}
+
+int s2n_server_name_send(struct s2n_connection *conn, struct s2n_stuffer *out)
+{
+    /* Write nothing. The extension just needs to exist. */
+    return S2N_SUCCESS;
+}
+
+int s2n_server_name_recv(struct s2n_connection *conn, struct s2n_stuffer *extension)
+{
+    notnull_check(conn);
+    /* Read nothing. The extension just needs to exist. */
+    conn->server_name_used = 1;
```

Comment:
I am a little confussed here, shouldnt we be receiving only if `conn->server_name_used` is set to 1?